packages/contracts/contracts/SmartInvoiceInstant.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "./interfaces/ISmartInvoiceInstant.sol";
+import "./interfaces/IWRAPPED.sol";
+
+contract SmartInvoiceInstant is
+    ISmartInvoiceInstant,
+    Initializable,
+    Context,
+    ReentrancyGuard
+{
+    using SafeERC20 for IERC20;
+
+    address public wrappedNativeToken;
+
+    address public client;
+    address public provider;
+    address public token;
+    bytes32 public details;
+    uint256 public deadline;
+
+    uint256[] public amounts;
+    uint256 public total = 0;
+    uint256 public totalFulfilled = 0;
+    bool public fulfilled;
+    uint256 public constant MAX_DEADLINE = 63113904; // 2-year limit on locker
+
+    event Deposit(address indexed sender, uint256 amount);
+    event Withdraw(address indexed recipient, uint256 amount);
+
+    // solhint-disable-next-line no-empty-blocks
+    function initLock() external initializer {}
+
+    function init(
+        address _recipient,
+        uint256[] calldata _amounts,
+        bytes calldata _data
+    ) external override initializer {
+        require(_recipient != address(0), "invalid provider");
+
+        _handleData(_data);
+
+        provider = _recipient;
+
+        uint256 _total = 0;
+        for (uint256 i = 0; i < _amounts.length; i++) {
+            _total += _amounts[i];
+        }
+        total = _total;
+    }
+
+    /**
+     * @dev calculates the total amount due. Extensible to include late fees, etc.
+     */
+    function getTotalDue() public view returns (uint256) {
+        return total;
+    }
+
+    // withdraw locker remainder to client if termination time passes & no lock
+    function withdraw() external override nonReentrant {
```

Comment:
this looks like it sends to the provider below; should this withdraw to client if this is a locker safeguard?

Reply:
No its intentionally meant to send to provider. Client "deposits" (pay) invoice and provider "withdraws" (collects) funds from contract. I wanted to use same syntax as much as possible but I could see how that led to some confusion. Would `collect()` or `release()` make more sense?

Reply:
ok; i think it was the comment the line above `withdraw locker remainder to client if termination time passes & no lock` that was confusing. Should there be another function for a client safeguard to withdraw if the termination time passes or is this just a leftover comment?

Reply:
It's just a leftover comment